Matomo Analytics Support

kenjibailly opened the pull request:
# Proposed Changes
<!--- Provide a general summary of your changes -->
In addition to Matomo Support #111 to track browsers with javascript disabled

- Image tracking tag in body

## Checklist

- [x] Tested locally
- [x] Ran `yarn ci` to test my code
- [x] Did not add any unnecessary changes
- [x] All my changes appear after other changes in each file
- [x] Included a screenshot (if adding a new button)
- [x] 🚀

<!--- If adding a new button, please include screenshot -->
<!--- If you are adding new code, please follow the pattern and add it to the end of the file where appropriate -->
<!--- Please be sure that you are not adding any additional changes including spaces, adding/deleting lines -->

timothystewart6 reviewed (changes requested):
I think we're missing some additional config for ENV variables that you had in your last PR?  We need to always configure the `RAZZLE_` ones too!

kenjibailly commented:
These env's are the same as for the other script, so these have already been initialized. Or am I not getting something here? I don't understand

timothystewart6 commented:
If they are not configured there we might not have access to them while debugging locally, although this is really only needed server side it's still nice to be consistent.

kenjibailly commented:
Isn't what this is for? 
Copied this from #111 
```
        MATOMO_URL: nodeIsProduction
          ? process.env.MATOMO_URL
          : process.env.RAZZLE_MATOMO_URL,
        MATOMO_SITE_ID: nodeIsProduction
          ? process.env.MATOMO_SITE_ID
          : process.env.RAZZLE_MATOMO_SITE_ID,
```

Also, I saw the commit had been applied from #111 yet I cannot see the changes in the files.
It's my second time contributing to a project, real github contribution noob here. Sorry

timothystewart6 commented:
No problem!  Yes, that's the change but it's in a PR that you closed.  I don't see those changes in this PR listed here https://github.com/techno-tim/littlelink-server/pull/112/files

You are so close, just open up your current PR and add those changes and push it up!

kenjibailly commented:
> No problem! Yes, that's the change but it's in a PR that you closed. I don't see those changes in this PR listed here https://github.com/techno-tim/littlelink-server/pull/112/files
> 
> You are so close, just open up your current PR and add those changes and push it up!

I didn't mean to do that! I had a problem where my first repository committed every file under my comment "matomo support" and therefore I deleted it and created a new one. Didn't know that would close #111 since I thought that was already pushed.

I think I have now added it again, do we have everything now? :)

timothystewart6 commented:
this closes https://github.com/techno-tim/littlelink-server/issues/41

timothystewart6 commented:
Thank you! Will review in the morning!

kenjibailly commented:
Thank you for bearing with me :)

timothystewart6 commented on src/server.js:
```diff
@@ -168,9 +168,41 @@ server
+          runtimeConfig.MATOMO_URL && runtimeConfig.MATOMO_SITE_ID
+            ? `
+            <!-- Matomo Image Tracker-->
+            <img referrerpolicy="no-referrer-when-downgrade" src="${runtimeConfig.MATOMO_URL}matomo.php?idsite=${runtimeConfig.MATOMO_SITE_ID}&amp;rec=1" style="border:0" alt="" />
```
one small thing here!  I think this would be 

`<img referrerpolicy="no-referrer-when-downgrade" src="${runtimeConfig.MATOMO_URL}/matomo.php?idsite=${runtimeConfig.MATOMO_SITE_ID}&amp;rec=1" style="border:0" alt="" />`

(missing `/`)  Otherwise your url would have to end in a trailing slash, which might be fine but also odd that you would have to know to end your site in a slash.  What do you think?

after this, looks good!

kenjibailly replied:
The reason I did this is because the slash is there as well in the other part of the config where it just asks for the url. I didn't test if the slash was necessary there. Second thing I was thinking is that they will probably check out their config with this same code cause don't know how else you can get the site ID. I was thinking if they copied the URL from there (which has trailing slash) it would then create errors.

I also thought the same like you that it's pretty confusing. I guess it's confusing depending from which point you view it and could be either way confusing. 

I'll add another commit to what you said. Maybe my logic is too far fetched.

timothystewart6 replied:
I see what you mean but I think this is cleaner for the person using the URL as a variable, keep the ugly stuff in code ;)  Thank you!

timothystewart6 commented:
> 

You're doing fine!  One small question / possible fix, after that looks good!

kenjibailly commented:
Fix has been committed :D

timothystewart6 commented:
Thank you for you contribution!
